src/linker/Linker/TypeReferenceExtensions.cs

```diff
@@ -375,5 +375,14 @@ public static bool IsSubclassOf (this TypeReference type, string ns, string name
 
 			return false;
 		}
+
+		public static TypeDefinition ResolveToMainTypeDefinition (this TypeReference type)
```

Comment:
This is probably worth a comment what the method is supposed to do (and when it's useful) - I still don't like the name, but I don't have a better alternative.